### ISSUE TITLE
When a cucumber scenario fails in a @Before method then serenity reporter should report it as failed instead of ignored

### ISSUE THREAD

**Issue by wieben** [serenity-bdd/serenity-core#2935]
When a cucumber scenario fails in a `@Before` method then serenity reporter should report it as failed instead of ignored.
@wakaleo fixed this issue back in 2018 and it somehow reappeared so I found his fix from back then which need a small change.
I think that in the method
```
io.cucumber.core.plugin.SerenityReporter#handleTestCaseFinished
```
the line
```
if (Status.FAILED.equals(event.getResult()) && noAnnotatedResultIdDefinedFor(event)) {
```
should be changed to
```
if (Status.FAILED.equals(event.getResult().getStatus()) && noAnnotatedResultIdDefinedFor(event)) {
```
so that if a cucumber before step fails then serenity reporter also reports the scenario as failing instead of ignored.
And probably a test case created so that the intended behavior of serenity reporter reporting on failing `@Before` methods is verified.
I realize this submitted issue could be a pull request, if my suggested fix gets in, then my next contribution might be an actual pull request... ;)

**Comment by wakaleo** [comment]
Well spotted. You will also needto cater for Status.ERROR and Status.COMPROMISED. I look forward to your PR!

**Comment by wakaleo** [comment]
Fixed in 3.3.10
